Paltineanu Rares-Mihai/curs9 - 3.10/main.c

```diff
@@ -0,0 +1,333 @@
+/**
+Read  from  the  keyboard  strings  of at  most 80  characters,  representing  integer  or  real  non-exponential numbers, separated by spaces.
+Compute the sum of the real numbers and of the integer numbers of each string, except the incorrect values.
+An incorrect value is a character string delimited by spaces, containing non-numeric characters, or having the length greater than 5.
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+char **allocMat(int n) {
+
+    char **mat = (char **)malloc(sizeof(char *) * n);
+
+    for(int i = 0; i < n; i++) {
+        mat[i] = (char *)malloc(sizeof(char) * 81);
+
+    }
+
+    return mat;
+}
+
+void readMat(char **mat, int n) {
+
+    for(int i = 0; i < n; i++) {
+        gets(mat[i]);
+    }
+}
+
+int verify(char *vect) {
+
+    int sizeString;
+
+    sizeString = strlen(vect);
+
+    for(int i = 0; i < sizeString; i++) {
+
+
+        if(   vect[i] != '0' && vect[i] != '1' && vect[i] != '2' && vect[i] != '3' && vect[i] != '4'
```

Comment:
Could it be that because they are all characters you could be just comparing a range, instead of writing so many `!=` ?

Reply:
i don't think the problem is from this function, the addition on integer numbers is working, i think the issue is in the function that adds real numbers

Reply:
I'm sure the problem is not here! I was just giving some code advice. 

Have you tried debugging?

Reply:
Also, regarding your gets/scanf issue, have you tried `getline()` ?

https://c-for-dummies.com/blog/?p=1112

Reply:
I'm not very familiar with debbuging and i tried a few minutes ago the getline(), but it was not working. Maybe we will check at the lab

Reply:
I'll make a video on debugging as well then 😄

Reply:
That surprises me! :))) I just don't know where am i supposed to look, because if i press on watch, it shows me the array, i don't know how to explain, basically, i don't know how to see the effect during the program, not only the effect at the end of the program

Reply:
Put breakpoints on all the lines you want the program to stop - even in loops! It will stop there

Reply:
https://www.youtube.com/watch?v=Jab1qj_QR8s
Until I do it